packages/amplify-cli-core/src/help/help-helpers.ts

```diff
@@ -0,0 +1,236 @@
+  });
+}
+
+function printGenericHelp(context: $TSContext, commandsInfo: Array<CommandInfo>, defaultNumTabs=1, extraTabLengthThreshold=5) {
+  printBodyText(TAG_LINE);
+  printer.blankLine();
+
+  printHeaderText(USAGE);
+  printBodyText(DEFAULT_INDENT + 'amplify <command> <subcommand> [flags]');
+  printer.blankLine();
+
+  printHeaderText(COMMANDS);
+  let commandRows: Array<[string, string]> = commandsInfo.map(commandObject => [commandObject.command, commandObject.commandDescription]);
+  printColumns(commandRows, DEFAULT_COLUMN_SEP_SPACING);
+  printer.blankLine();
+
+  printHeaderText(FLAGS);
+  printBodyText(DEFAULT_INDENT + '-h' + FLAG_DELIMITER + '--help' + DEFAULT_COLUMN_SEP + 'Show help for a command');
+  printer.blankLine();
+
+  printHeaderText(LEARN_MORE);
+  printBodyText(DEFAULT_INDENT + 'Visit ' + DEFAULT_LINK);
+  printer.blankLine();
+}
+
+function printCommandSpecificHelp(context: $TSContext, commandsInfo: Array<CommandInfo>, commandName: string, defaultNumTabs=1, extraTabLengthThreshold=5) {
+  let command = lookUpCommand(commandsInfo, commandName.toLocaleLowerCase());
+  if (command === null) {
+    printGenericHelp(context, commandsInfo, defaultNumTabs, extraTabLengthThreshold);
+    return;
+  }
+
+  printHeaderText(USAGE);
+  printBodyText(DEFAULT_INDENT + command.commandUsage);
+
+  printer.blankLine();
+  if (command.commandUsage.includes('<category>')){
+    printCategoryMessage();
+  }
+
+  printHeaderText(DESCRIPTION);
+  printBodyText(DEFAULT_INDENT + command.commandDescription);
+  printer.blankLine();
+
+  if (command.subCommands.length > 0) {
+    printHeaderText(SUBCOMMANDS);
+    let subCommandRows: Array<[string, string]> = command.subCommands.map(subCommandObject => [subCommandObject.subCommand, subCommandObject.subCommandDescription]);
+    printColumns(subCommandRows, DEFAULT_COLUMN_SEP_SPACING);
+    printer.blankLine();
+    let subcommandOrCategory = command.commandUsage.includes('<category>') ? '<category>' : '<subcommand>';
+    printBodyText(DEFAULT_INDENT + 'Use "amplify ' + command.command + ' ' + subcommandOrCategory + ' -h" to see subcommand-specific help');
+    printer.blankLine();
+  }
+
+  if (command.commandFlags.length > 0) {
+    printHeaderText(FLAGS);
+    let flagRows: Array<[string, string]> = command.commandFlags.map(flagObject => getHelpFlagRow(flagObject));
+    printColumns(flagRows, DEFAULT_COLUMN_SEP_SPACING);
+    printer.blankLine();
+  }
+
+  if (command.learnMoreLink.length > 0) {
+    printHeaderText(LEARN_MORE);
+    printBodyText(DEFAULT_INDENT + command.learnMoreLink);
+    printer.blankLine();
+  }
+}
+
+function printSubcommandSpecificHelp(context: $TSContext, commandsInfo: Array<CommandInfo>, commandName: string, subcommandName: string, defaultNumTabs=1, extraTabLengthThreshold=5) {
+  let subCommand = lookUpSubcommand(commandsInfo, commandName, subcommandName);
+  if (subCommand === null) {
+    let command = lookUpCommand(commandsInfo, commandName);
```

Comment:
nit(here and elsewhere): I prefer `const` instead of `let` if you aren't reassigning the variable.